livedata/src/main/java/io/getstream/chat/android/livedata/utils/DefaultRetryPolicy.kt

```diff
@@ -2,14 +2,15 @@ package io.getstream.chat.android.livedata.utils
 
 import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.errors.ChatError
+import io.getstream.chat.android.livedata.extensions.isPermanent
 
 internal class DefaultRetryPolicy : RetryPolicy {
     override fun shouldRetry(
         client: ChatClient,
         attempt: Int,
         error: ChatError
     ): Boolean {
-        return attempt < 3
+        return attempt < 3 && !error.isPermanent()
     }
 
     /**
```

Comment:
Perhaps we should check on the error.isPermanent flag in runAndRetry? 

what point is there in allowing customers to retry permanent errors..?

Reply:
Moved the check to `runAndRetry`